rts/Lua/LuaSyncedRead.cpp

```diff
@@ -2960,6 +2905,37 @@ void ApplyPlanarTeamError(lua_State* L, int allegiance, float3& mins, float3& ma
 	}
 }
 
+/**
+ * @brief This populates the Lua table with the unit IDs of units that are in a
+ * particular region. 
+ * 
+ * @param allegiance The allegiance of the units to add to the table.
+ * @param units The units to add to the table.
+ * @param inRegion A lambda checking if the unit position is in the region
+ */
+template<typename InRegion>
+static void GetFilteredUnits(lua_State *L, int allegiance, const std::vector<CUnit *> &units, InRegion inRegion, bool fullRead) {
+	unsigned int count = 0;
+
+	const int readTeam = CLuaHandle::GetHandleReadTeam(L);
+	const int readAllyTeam = CLuaHandle::GetHandleReadAllyTeam(L);
+
+	auto isDisqualified = GetIsUnitDisqualifiedTest(L, allegiance, readTeam, readAllyTeam);
+
+	for(const CUnit *unit: units) {
+		if(isDisqualified(unit))
+			continue;
+
+		// The unit error will be zero if the unit is on an allied team
```

Comment:
Comment somewhat redundant, error being zero for allies is unsurprising. No need to add it just because the previous code explicitly skipped it